lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -370,13 +437,39 @@ struct ClassOpSignatureConversion : public OpConversionPattern<om::ClassOp> {
   }
 };
 
+struct ClassExternOpSignatureConversion
+    : public OpConversionPattern<om::ClassExternOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(om::ClassExternOp classOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Block *body = classOp.getBodyBlock();
+    TypeConverter::SignatureConversion result(body->getNumArguments());
+
+    // Convert block argument types.
+    if (failed(typeConverter->convertSignatureArgs(body->getArgumentTypes(),
+                                                   result)))
+      return failure();
+
+    // Convert the body.
+    if (failed(rewriter.convertRegionTypes(body->getParent(), *typeConverter,
```

Comment:
Can these have bodies? Presumably it's fine/safe to do this anyway?

Reply:
Actually, `om.class.extern` ops do have bodies, consisting of `om.class.extern.field` ops... this is how we convert the types of fields :+1: